calculator.rb

```diff
@@ -0,0 +1,57 @@
+# CALCULATOR
+# Program accepts user input to perform calculator evaluations.
+
+# TODO: 
+# -Write the actual evaluation function 
+# -refactor validate code 
+
+
+operators = { 
+  :addition => ["+", "plus", "add"],
+  :subtraction => ["-", "minus", "subtract"],
+  :multiplication => ["*", "times","multiply"],
+  :division => ["/", "divide","split"],
+  :mod => ["%", "mod","modulo"],
+  :exponentialize => ["^", "to the power of","exponential"]
+}
+
+
+puts "CALCULATOR"
+puts "**********\n\n"
+
+puts "Welcome to the CALCULATOR. Available operators include:\n\n"
+
+#pretty-print the list of valid operators 
+operators.each do |k,v|
+  puts "#{k}: #{v.join(", ")}"
+end
+
+
+equation = {
+  "operator" => nil,
+  "left number" => nil,
+  "right number" => nil,
+}
+
+equation.each do |equation_piece, value|
+  print "Please give me the #{equation_piece.to_s} > "
+  input = gets.strip
+
+  if equation_piece == "operator"
+    until operators.all? {|operation| operation.has} #TO-DO: figure out how to check input against all data in the operator array to find match
```

Comment:
There is no method `.has` on the operation

You are also missing an `end` to this `until` loop.